app/controllers/users_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.create(user_params)
+    if @user.save
+      redirect_to new_user_path
+    else
+      render :new, status: :unprocessable_entity # important to return the error message
+
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      redirect_to new_user_path
```

Comment:
- [ ] If the user is saved after update, the redirect should be to the edited user.i.e
```ruby
redirect_to edit_user_path
```